src/checks/quote_character.rs

```diff
@@ -98,6 +98,16 @@ mod tests {
         run_quote_char_tests(asserts);
     }
 
+    #[test]
+    fn with_dollar_sign_test() {
+        let asserts = vec![
+            (line_entry(1, 3, "FOO=BAR"), None),
+            (line_entry(3, 3, "FOO=$BAR BAR"), None),
```

Comment:
It's necessary to check for cases where the `$` character exists but quotes do not cause a `Warning` 🙂 

```suggestion
            (line_entry(1, 2, "BAR=\"$ABC\""), None),
            (line_entry(2, 2, "FOO='${BAR}BAR'"), None),
```